includes/class-sensei-frontend.php

```diff
@@ -849,17 +849,26 @@ private function maybe_redirect_to_next_lesson( $lesson_id = 0 ) {
 			return;
 		}
 
-		$nav_links = sensei_get_prev_next_lessons( $lesson_id );
+		$nav_links    = sensei_get_prev_next_lessons( $lesson_id );
+		$redirect_url = false;
 
 		if ( isset( $nav_links['next'] ) ) {
-			/**
-			 * Filter the URL that students are redirected to after completing a lesson.
-			 *
-			 * @since 1.12.0
-			 *
-			 * @param string $redirect_url URL to redirect students to after completing a lesson.
-			 */
-			wp_safe_redirect( apply_filters( 'sensei_complete_lesson_redirect_url', esc_url_raw( $nav_links['next']['url'] ) ) );
+			$redirect_url = $nav_links['next']['url'];
+		}
+
+		/**
+		 * Filter the URL that students are redirected to after completing a lesson.
+		 *
+		 * @since 1.12.0
+		 *
+		 * @param string|bool $redirect_url URL to redirect students to after completing a lesson. False to skip redirect.
+		 * @param int         $lesson_id    Current lesson ID.
+		 * @param array       $nav_links    Navigation links found for the current lesson.
+		 */
+		$redirect_url = apply_filters( 'sensei_complete_lesson_redirect_url', $redirect_url, $lesson_id, $nav_links );
```

Comment:
I'm not sure if this is a big deal, but technically this is not backwards compatible. Previously we were handing in the escaped URL, and now we're handing it in unescaped and escaping it later.

I think this way is better, though. I wonder if there are any problems that people might run into with existing code? I can't think of anything...

Reply:
I think we should be escaping post-filter. I also can’t think of anything (besides 🙈 things) that this would break.